SQLAlchemy/DDL: Allow to turn off column store

fetzerms opened the pull request:
### Summary of the changes / Why this is an improvement

This allows to disable column store for a text column using:
```python
my_data: str = Column(String, crate_index=False, crate_columnstore=False)
```

Which will yield:
```sql
 "my_data" TEXT INDEX OFF STORAGE WITH ( columnstore = false )
```


## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

amotl commented:
Dear Matthias,

thanks a stack for submitting this improvement to the DDL compiler. In order to make the patch complete, we will need to add software tests and documentation. Would you be up to add them, or do you want us to take over?

With kind regards,
Andreas.

fetzerms commented:
@amotl Thank you very much. I will add some tests and documentation - just not sure when, as I'm quite occupied at work lately. But as the changes are very minor, I think I will soon push some changes.

fetzerms commented:
@amotl I have added the tests and some (little) docs. If I got it right, columnstore can only be turned off for TEXT columns, hence I added this as a condition to the code. In case this needs some rework, please let me know.

amotl reviewed (approved):
Dear Matthias,

your patch looks good to me. Thank you very much for adding corresponding software tests.

With kind regards,
Andreas.

amotl commented on src/crate/client/sqlalchemy/tests/create_table_test.py:
```diff
@@ -232,3 +232,30 @@
             self.Base.metadata.create_all(bind=self.engine)
+
+    def test_text_column_without_columnstore(self):
+        class DummyTable(self.Base):
```
We know this admonition by CodeQL on those occasions. They can be dismissed as **false positive**.

-- https://github.com/github/codeql/issues/11407

amotl replied:
Hi again. CodeQL admonition items will be made collapsible, it's so sweet. Thank you, @anaarmas.

-- https://github.com/github/codeql-action/issues/1411#issuecomment-1580176191

amotl commented:
Dear Matthias,

after adding a changelog item to your patch, I've just merged it. Thank you again.

With kind regards,
Andreas.
